CmdStarter.Lib/Attributes/AutoComplete/AutoCompleteAttribute.cs

```diff
@@ -0,0 +1,100 @@
+﻿using com.cyberinternauts.csharp.CmdStarter.Lib.Interfaces;
+using System.CommandLine.Completions;
+
+namespace com.cyberinternauts.csharp.CmdStarter.Lib.Attributes
+{
+    /// <summary>
+    /// Defines auto-completions for an <see cref="System.CommandLine.Option"/> or an <see cref="System.CommandLine.Argument"/>.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Parameter, AllowMultiple = true)]
+    public class AutoCompleteAttribute : Attribute
+    {
+        const string NULL_OR_EMPTY_ERROR_MESSAGE = "A completion cannot be null or empty!";
+
+        protected readonly string[] _labels;
+        protected LinkedList<CompletionItem>? _items;
```

Comment:
At the moment they need to be `protected` because of the overridden `CacheItems()`.

After you answered to the comment above (relating to that) we can discuss this.

Reply:
If you want to keep the possibility to override, then let them `protected`

Reply:
Made them private.